scripts/pr_force_push_report/force_push_report.py

```
@@ -0,0 +1,285 @@
+    Args:
+        response: Parsed JSON response from GitHub GraphQL API
+
+    Returns:
+        List of ForcePushEvent objects sorted chronologically (oldest first)
+    """
+    events: list[ForcePushEvent] = []
+
+    try:
+        nodes = response["data"]["repository"]["pullRequest"]["timelineItems"]["nodes"]
+    except (KeyError, TypeError) as exc:
+        LOGGER.warning(f"Unexpected response structure: {exc}")
```

Ruff (0.15.6)

[warning] Logging statement uses f-string

(G004)